app/graphql/mutations/customers/update.rb

```diff
@@ -21,7 +21,8 @@ class Update < BaseMutation
       type Types::Customers::Object
 
       def resolve(**args)
-        result = ::Customers::UpdateService.new(context[:current_user]).update(**args)
+        customer = context[:current_user].customers.find_by(id: args[:id])
+        result = ::Customers::UpdateService.call(customer:, args:)
```

Comment:
Just a thought, but could we leverage `raise_if_error` here as well?

Reply:
@nudded 

It would require a change in the way we treat the result:

```ruby
def resolve(**args)
  ::Customers::UpdateService.call(customer:, args:).raise_if_error!.customer
rescue BaseService::FailedResult
  result_error(result)
end
```

It's a personal opinion, but I don't feel it makes it easier to read.

Reply:
oh seeing it written down like that I tend to agree @vincent-pochet 👍 (It would only make sense if we captured that error somewhere generically perhaps. But agreed to keep this PR small :)